src/ranged.cpp

```diff
@@ -266,26 +263,43 @@ bool player::handle_gun_damage( item &it )
         }
     }
     if( curammo_effects.count( "BLACKPOWDER" ) ) {
-        if( !it.faults.count( fault_gun_blackpowder ) &&
-            it.faults_potential().count( fault_gun_blackpowder ) ) {
-            it.faults.insert( fault_gun_blackpowder );
-        }
         if( ( it.ammo_data()->ammo->recoil < firing->min_cycle_recoil ) &&
-            it.faults_potential().count( fault_gun_chamber_spent ) ) {
+            it.faults_potential().count( fault_gun_chamber_spent ) && ( it.type->gun->ups_charges < 1 ) ) {
```

Comment:
supposed to prevent a cycling failure in guns requiring external power despite the use of blackpowder. Blackpowder's different burning rate would only make a gas-operated gun fail to cycle.

Reply:
Understood, thanks